src/waypoint-hoc.jsx

```diff
@@ -36,6 +40,9 @@ const waypoint = Component => {
     constructor(props) {
       super(props);
       this.state = { scrolled: {} };
+      this._handleScroll = (this.props.throttle ?
+        throttle(this._handleScroll, this.props.throttle) :
+        this._handleScroll).bind(this);

```

Comment:
Throttled scroll listeners is recommended all over the place, 
Ex:
* http://ejohn.org/blog/learning-from-twitter/
* http://benalman.com/code/projects/jquery-throttle-debounce/examples/throttle/

By default handleScroll is not throttled.
I agree this need more test and doc.